src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Tooltip/DefaultTagHelperTooltipFactory.cs

```diff
@@ -381,5 +403,252 @@ private MarkupKind GetMarkupKind()
+                            var reducedName = ReduceTypeName(currentRunText.ToString());
+                            ClassifyTextAsKeywordOrType(runs, reducedName);
+                        }
+                        else
+                        {
+                            ClassifyTextAsKeywordOrType(runs, currentRunText.ToString());
+                        }
+
+                        currentRunText.Clear();
+                    }
+
+                    runs.Add(new RazorClassifiedTextRun(RazorPredefinedClassificationTypeNames.Punctuation, ch.ToString()));
+                }
+                else
+                {
+                    currentRunText.Append(ch);
+                }
+            }
+
+            if (currentRunText.Length != 0)
+            {
+                ClassifyTextAsKeywordOrType(runs, currentRunText.ToString());
+            }
+        }
+
+        private static void ClassifyTextAsKeywordOrType(List<RazorClassifiedTextRun> runs, string typeName)
+        {
+            var nullableType = typeName.EndsWith("?");
+            if (nullableType)
+            {
+                // Classify the '?' symbol separately from the rest of the type since it's considered punctuation.
+                typeName = typeName.Substring(0, typeName.Length - 1);
+            }
+
+            // Case 1: Type can be aliased as a C# built-in type (e.g. Boolean -> bool, Int32 -> int, etc.).
+            if (TypeNameToAlias.TryGetValue(typeName, out var aliasedTypeName))
+            {
+                runs.Add(new RazorClassifiedTextRun(RazorPredefinedClassificationTypeNames.Keyword, aliasedTypeName));
+            }
+            // Case 2: Type is a C# built-in type (e.g. bool, int, etc.).
+            else if (CSharpBuiltInTypes.Contains(typeName))
+            {
+                runs.Add(new RazorClassifiedTextRun(RazorPredefinedClassificationTypeNames.Keyword, typeName));
+            }
+            // Case 3: All other types.
+            else
+            {
+                var typeNameParts = typeName.Split('.');
+
+                var partIndex = 0;
+                foreach (var typeNamePart in typeNameParts)
+                {
+                    if (partIndex != 0)
+                    {
+                        runs.Add(Dot);
+                    }
+
+                    runs.Add(new RazorClassifiedTextRun(RazorPredefinedClassificationTypeNames.Type, typeNamePart));
+                    partIndex++;
+                }
+            }
+
+            if (nullableType)
+            {
+                runs.Add(NullableType);
+            }
+        }
+
+        private static bool TryClassifySummary(List<RazorClassifiedTextRun> runs, string documentation)
+        {
+            if (!TryExtractSummary(documentation, out var summaryContent))
+            {
+                return false;
+            }
+
+            runs.Add(NewLine);
+
+            var finalSummaryContent = CleanSummaryContent(summaryContent);
```

Comment:
Ah, I think you should probably build your own `clean` because this converts from XML summary bits into `Markdown`. If you built your own you could go directly from XML summary bits into classified items, it'd probably save loads of allocations too